core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/services/categories.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Observable } from 'rxjs';
+
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { inject, Injectable } from '@angular/core';
+
+import { pluck } from 'rxjs/operators';
+
+import { DotCMSResponse } from '@dotcms/dotcms-js';
+
+import { DotCategoryFieldCategory } from '../models/dot-category-field.models';
+
+export const API_URL = '/api/v1/categories';
+
+export const ITEMS_PER_PAGE = 7000;
+
+/**
+ * CategoriesService class.
+ *
+ * This service is responsible for retrieving the children of a given inode.
+ */
+@Injectable()
+export class CategoriesService {
+    readonly #http: HttpClient = inject(HttpClient);
```

Comment:
@oidacra Here, the type `HttpClient` is not necessary because the `inject(HttpClient)` inference will determine the correct type.

Reply:
Actually, I think we have a lint for that.